fdb-sql-layer-core/src/main/java/com/foundationdb/server/types/common/funcs/ReadBlob.java

```
@@ -0,0 +1,128 @@
+            if (o instanceof BlobRef) {
+                blob = (BlobRef) o;
+                String mode = context.getQueryContext().getStore().getConfig().getProperty(AkBlob.RETURN_UNWRAPPED);
+                if (mode.equalsIgnoreCase(AkBlob.UNWRAPPED)) {
+                    data = blob.getBytes();
+                    outputData = getSubArray(data, offset, length);
+                } else {
+                    if (blob.isShortLob()) {
+                        data = blob.getBytes();
+                        outputData = getSubArray(data, offset, length);
+                    } else {
+                        LobService ls = context.getQueryContext().getServiceManager().getServiceByClass(LobService.class);
+                        outputData = ls.readBlob(context.getQueryContext().getSession(), blob.getId(), offset, length);
+                    }
+                }
+            } else if (o instanceof UUID) {
+                LobService ls = context.getQueryContext().getServiceManager().getServiceByClass(LobService.class);
+                outputData = ls.readBlob(context.getQueryContext().getSession(), (UUID)o, offset, length);
+            } else {
+                try {
+                    UUID id = UUID.fromString(inputs.get(0).getString());
+                    LobService ls = context.getQueryContext().getServiceManager().getServiceByClass(LobService.class);
+                    outputData = ls.readBlob(context.getQueryContext().getSession(), id, offset, length);
+                } catch (IllegalArgumentException iea) {
+                    throw new LobNotFoundException("Format of blob id does not match a GUID format");
+                }
+            }
+        }
+        if (outputData == null){
+            outputData = new byte[0];
+        }
+        output.putBytes(outputData);
+    }
+
+    private byte[] getSubArray(byte[] data, long offset, int length) {
+        if (offset > Integer.MAX_VALUE) {
+            return null;
```

An exception might be better than silently returning nothing.